File: hw/fuentes/U3/HW17Maze/MazeAppPY/ec/edu/espe/Maze/view/MazePrinter.py
def print_maze_ascii(maze):
    N = maze.N
    M = maze.M
    rooms = maze.get_rooms()

    for i in range(N):

        for j in range(M):
            room = rooms[i][j]
            if room.has_wall('N'):
                print("+---", end="")
            else:
                print("+   ", end="")
        print("+")

        for j in range(M):
            room = rooms[i][j]
            if room.has_wall('W'):
                print("|", end="")
            else:
                print(" ", end="")

            if room.get_door():
                door_type = room.get_door().get_type()
                if door_type == "Entrance":
                    print(" E ", end="")
                elif door_type == "Exit":
                    print(" X ", end="")
                else:
                    print("   ", end="")
            else:
                print("   ", end="")

        print("|" if rooms[i][-1].has_wall('E') else " ")

    for j in range(M):
        if rooms[-1][j].has_wall('S'):
            print("+---", end="")
        else:
            print("+   ", end="")
    print("+")

File: hw/fuentes/U3/HW17Maze/NewMazeGenerator/src/main/java/ec/edu/espe/mazegenerator/view/MazePrinter.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.mazegenerator.view;

import ec.edu.espe.mazegenerator.model.EntranceDoor;
import ec.edu.espe.mazegenerator.model.ExitDoor;
import ec.edu.espe.mazegenerator.model.Maze;
import ec.edu.espe.mazegenerator.model.Room;

/**
 *
 * @author jorge
 */
public class MazePrinter {
    public static void printMazeAscii(Maze maze) {
        Room[][] rooms = maze.getRooms();
        int N = Maze.N;
        int M = Maze.M;

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < M; j++) {
                Room room = rooms[i][j];
                if (hasWall(room, 'N')) {
                    System.out.print("+---");
                } else {
                    System.out.print("+   ");
                }
            }
            System.out.println("+");

            for (int j = 0; j < M; j++) {
                Room room = rooms[i][j];
                if (hasWall(room, 'W')) {
                    System.out.print("|");
                } else {
                    System.out.print(" ");
                }

                if (room.getDoor() instanceof EntranceDoor) {
                    System.out.print(" E ");
                } else if (room.getDoor() instanceof ExitDoor) {
                    System.out.print(" X ");
                } else {
                    System.out.print("   ");
                }
            }

            System.out.println("|");
        }

        for (int j = 0; j < M; j++) {
            if (hasWall(rooms[N - 1][j], 'S')) {
                System.out.print("+---");
            } else {
                System.out.print("+   ");
            }
        }
        System.out.println("+");
    }

    public static boolean hasWall(Room room, char direction) {
        return room.hasWall(direction);
    }
    
}
